code/game/turfs/turf.dm

```diff
@@ -105,6 +112,28 @@ GLOBAL_LIST_EMPTY(station_turfs)
 /turf/proc/Initalize_Atmos(times_fired)
 	CALCULATE_ADJACENT_TURFS(src)
 
+/turf/examine(mob/user)
+	. += ..()
+	if(!damage_amount)
+		. += "<span class='notice'>It looks fully intact.</span>"
+	else
+		var/dam = round((heat_integrity/max_heat_integrity)* 100, 1)
+		. += "<span class='warning'>The remaining integrity is at [dam]%</span>"
+
+/turf/welder_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(user.a_intent == INTENT_GRAB)
```

Comment:
This is pretty confusing choice. Also fixing all types of walls with a welder seems off.

Reply:
the idea here was to give a way to fix the turfs in a quick way.